### PR TITLE
Feature/pt 995 openapi

### PR DESCRIPTION
**Pull request by ydrisR**
Add open api documentation headers + script to build the swagger doc during build

Commits to review: from "Create a preview service for hydrophone email templates" to "Build api documentation during build"

### CODE REVIEW

File: api/verify.go [hunk 1]
```diff
@@ -12,10 +12,16 @@ const (
...
-// Send a test email to prove all configuration is in place for sending emails
-//
-// status: 200
-// status: 500 internal server error
+// @Summary Sanity check the service configuration
+// @Description  Send a test email to prove all configuration is in place for sending emails
+// @ID hydrophone-api-sanityCheck
+// @Accept  json
+// @Produce  json
+// @Param userID path string true "user id"
```
Comment (yvon-dblg) [inline api/verify.go]: Error from swagger:

> Parameter names are case-sensitive. The parameter named "userID" does not match the case used in the path "/sanity_check/{userId}".

File: api/hydrophoneApi.go [hunk 2]
```diff
@@ -175,6 +174,14 @@ func (h varsHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 	h(res, req, vars)
 }
 
+// @Summary Get the api status
+// @Description Get the api status
+// @ID hydrophone-api-getstatus
+// @Accept  json
+// @Produce  json
+// @Success 200
```
Comment (yvon-dblg) [inline api/hydrophoneApi.go]:
```suggestion
// @Success 200 {string} "OK"
```
Swagger error:

> should have required property 'description' missingProperty: description

File: api/signup.go [hunk 3]
```diff
@@ -149,17 +159,21 @@ func (a *Api) sendSignUpInformation(res http.ResponseWriter, req *http.Request,
...
+// @ID hydrophone-api-sendSignUp
+// @Accept  json
+// @Produce  json
+// @Param userid path string true "user id"
+// @Success 200 {string} string "OK"
+// @Failure 400 {object} status.Status "userId was not provided, return:\"Required userid is missing\" "
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provided sufficient privileges"
+// @Failure 403 {object} status.Status "Operation is forbiden, return detailed error"
+// @Failure 500 {object} status.Status "Internal error while processing the confirmation, detailled error returned in the body"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
```
Comment (yvon-dblg) [inline api/signup.go]:
```suggestion
// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service)"
```

File: api/signup.go [hunk 4]
```diff
@@ -290,11 +304,18 @@ func (a *Api) sendSignUp(res http.ResponseWriter, req *http.Request, vars map[st
...
+// @Summary Resend a signup confirmation email to a user who have not confirmed yet
+// @Description  If a user didn't receive the confirmation email and logs in, they're directed to the confirmation-required page which can
+// @Description  offer to resend the confirmation email.
+// @ID hydrophone-api-resendSignUp
+// @Accept  json
+// @Produce  json
+// @Param useremail path string true "user email address"
+// @Success 200 {string} string "OK"
+// @Failure 404 {object} status.Status "Confirmation not found, return \"No matching signup confirmation was found\" "
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
```
Comment (yvon-dblg) [inline api/signup.go]:
```suggestion
// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service)"
```

File: api/signup.go [hunk 5]
```diff
@@ -363,18 +384,22 @@ func (a *Api) resendSignUp(res http.ResponseWriter, req *http.Request, vars map[
...
+// @ID hydrophone-api-acceptSignUp
+// @Accept  json
+// @Produce  json
+// @Param confirmationid path string true "confirmation id"
+// @Param details body models.Acceptance false "new password (optional)"
+// @Success 200 {string} string "OK"
+// @Failure 400 {object} status.Status "confirmationid was not provided, return:\"Required confirmation id is missing\" "
+// @Failure 404 {object} status.Status "Confirmation not found or expired"
+// @Failure 409 {object} status.Status "Payload is missing or invalid. Return the detailled error"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
```
Comment (yvon-dblg) [inline api/signup.go]:
```suggestion
// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service)"
```

File: hydrophone.go [hunk 6]
```diff
@@ -1,3 +1,20 @@
+// @title Hydrophone API
+// @version 0.0.1
```
Comment (yvon-dblg) [inline hydrophone.go:2]: Should we use the v0.0.1 for the API description ?

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
There is a few minor fix todo.

Maybe add to .gitignore:
- docs/docs.go
- docs/swagger.json
- docs/swagger.yaml
?

Nice work :-)

File: api/verify.go (hunk 1 above)
Comment (ydrisR) [reply api/verify.go]: ok

File: api/hydrophoneApi.go (hunk 2 above)
Comment (ydrisR) [reply api/hydrophoneApi.go]: OK

File: api/signup.go (hunk 3 above)
Comment (ydrisR) [reply api/signup.go]: ok

File: hydrophone.go (hunk 6 above)
Comment (ydrisR) [reply hydrophone.go:2]: Good question, to discuss with the team

**Review by yvon-dblg — APPROVED** [review]

**Pull request merged**